expression/builtin_other.go

```diff
@@ -103,14 +135,53 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 	return sig, nil
 }
 
+type baseInSig struct {
+	baseBuiltinFunc
+	nonConstArgs []Expression
+	hasNull      bool
+}
+
 // builtinInIntSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInIntSig struct {
-	baseBuiltinFunc
+	baseInSig
+	hashSet map[int64]bool
+}
+
+func (b *builtinInIntSig) buildHashMapForConstArgs(ctx sessionctx.Context) error {
+	b.nonConstArgs = []Expression{b.args[0]}
+	b.hashSet = make(map[int64]bool, len(b.args)-1)
+	count := 0
```

Comment:
no need to maintain the length of `map` manually. you can obtain this information by calling `len(b.hashSet)`.